### PR TITLE
[HDF5] Update version

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

File: H/HDF5/build_tarballs.jl [hunk 1]
```diff
@@ -11,21 +11,21 @@ sources = [
...
+    ArchiveSource("http://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-hdf5-1.12.0-1-any.pkg.tar.xz", "0730467750257fe103667d9db0b28357e26258ca86aad868f5af4e8020ec3e04"),
     ArchiveSource("http://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-szip-2.1.1-2-any.pkg.tar.xz", "ec8fe26370b0673c4b91f5ccf3404907dc7c24cb9d75c7b8830aa93a7c13ace7"),
-    ArchiveSource("http://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-hdf5-1.10.5-1-any.pkg.tar.xz", "e01196dd53711304aa4026932c153171606efc4d6938dd3c172b6b40d9e7cdd9"),
     ArchiveSource("http://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-zlib-1.2.11-7-any.pkg.tar.xz", "1decf05b8ae6ab10ddc9035929014837c18dd76da825329023da835aec53cec2"),
 
      # We need some special compiler support libraries from mingw
-    ArchiveSource("http://repo.msys2.org/mingw/i686/mingw-w64-i686-gcc-libs-9.1.0-3-any.pkg.tar.xz", "416819d44528e856fb1f142b41fd3b201615d19ddaed8faa5d71296676d6fa17"),
+    ArchiveSource("http://repo.msys2.org/mingw/i686/mingw-w64-i686-gcc-libs-10.2.0-1-any.pkg.tar.zst", "104994c697fe1e03a85b3d82bd7642a0b18512e41eb4d565bb42bdec52b4bbec"),
```
Comment (musm) [inline H/HDF5/build_tarballs.jl:22]: Not sure if `zst` is supported

Comment (musm) [reply H/HDF5/build_tarballs.jl]: Edit, it's not

**Comment by giordano** [comment]
I don't understand this pull request, we have different sources corresponding to different versions of HDF5?

**Comment by musm** [comment]
That's always been the case, unfortunately, until HDF5 offers cross compilation. I'm just bumping up the versions available.

**Comment by visr** [comment]
Will this improve or worsen #1112? Seems like before there was 1.10.4 and 1.10.5, and now 1.12 is added?

**Comment by musm** [comment]
It doesn't worsen it, it's already mismatched :(

**Comment by musm** [comment]
I think we have been waiting for HDF5 to add cross-compilation for the last 4 years, I doubt they will ever get to it.

**Comment by visr** [comment]
Do you think this may help the situation? https://github.com/JuliaPackaging/Yggdrasil/issues/1112#issuecomment-659230515

**Comment by musm** [comment]
Well clearly this doesn't help and makes the version mismatch worse. We can close this issue. I don't know if it's a good idea to try cross-compilation with wine for Windows, that might be more trouble that it's worth and I'm not sure the benefits it has.

**Comment by visr** [comment]
Yeah, not a great situation :(

**Pull request closed without merging**